### PR TITLE
More fixes regarding spark/hive identifiers

### LINKED ISSUE #950: [spark dialect] Parser still doesn't work when column name/alias starts with digits AND contains underscore
hi, struggling with spark sql scripts, I found https://github.com/tobymao/sqlglot/pull/930 still not fix bugs with name/alias startswith digits and conatain underscore. as exsample below 

```python
>>> sql="select 1a_1a from test_a"
>>> parse_one(sql, read='spark')
(SELECT expressions: 
  (ALIAS this: 
    (COLUMN this: 
      (IDENTIFIER this: 1a_, quoted: False)), alias: 
    (IDENTIFIER this: 1a, quoted: False)), from: 
  (FROM expressions: 
    (TABLE this: 
      (IDENTIFIER this: test_a, quoted: False))))

>>> sql="select 1a as 1a_1a from test_a"
>>> parse_one(sql, read='spark')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/opt/homebrew/lib/python3.8/site-packages/sqlglot/__init__.py", line 84, in parse_one
    result = dialect.parse(sql, **opts)
  File "/opt/homebrew/lib/python3.8/site-packages/sqlglot/dialects/dialect.py", line 150, in parse
    return self.parser(**opts).parse(self.tokenizer.tokenize(sql), sql)
  File "/opt/homebrew/lib/python3.8/site-packages/sqlglot/parser.py", line 670, in parse
    return self._parse(
  File "/opt/homebrew/lib/python3.8/site-packages/sqlglot/parser.py", line 735, in _parse
    self.raise_error("Invalid expression / Unexpected token")
  File "/opt/homebrew/lib/python3.8/site-packages/sqlglot/parser.py", line 778, in raise_error
    raise error
sqlglot.errors.ParseError: Invalid expression / Unexpected token. Line 1, Col: 17.
  select 1a as 1a_1a from test_a
```

which are both valid in spark sql: 
```python
>>> spark.sql("create table test_table (1a string)")
DataFrame[]
>>> spark.sql("select 1a as 1a_1a from test_table")
DataFrame[1a_1a: string]
>>> spark.sql("create table test_table2 (1a_1a string)")
DataFrame[]
>>> spark.sql("select 1a_1a from test_table2")
DataFrame[1a_1a: string]
```

### PR DESCRIPTION
**Pull request by GeorgeSittas**
fixes #950

### CODE REVIEW

File: sqlglot/tokens.py [hunk 1]
```diff
@@ -947,9 +947,12 @@ def _scan_number(self) -> None:
             elif self._peek.isidentifier():  # type: ignore
                 number_text = self._text
                 literal = []
-                while self._peek.isidentifier():  # type: ignore
-                    literal.append(self._peek.upper())  # type: ignore
+
+                char = self._peek.strip()  # type: ignore
```
Comment (tobymao) [inline sqlglot/tokens.py:951]: why do you need to strip

Comment (GeorgeSittas) [reply sqlglot/tokens.py:951]: Because we want to stop tokenizing a possible VAR if we encounter whitespace.

Comment (tobymao) [reply sqlglot/tokens.py:951]: can you do 

while char.strip() and …

Comment (GeorgeSittas) [reply sqlglot/tokens.py]: Done.

**Review by tobymao — APPROVED** [review]

**Pull request merged**